src/timedata.cpp

```diff
@@ -40,16 +40,20 @@ static int64_t abs64(int64_t n)
     return (n >= 0 ? n : -n);
 }
 
+#define BITCOIN_TIMEDATA_MAX_SAMPLES 200
+
 void AddTimeData(const CNetAddr& ip, int64_t nOffsetSample)
 {
     LOCK(cs_nTimeOffset);
     // Ignore duplicates
     static set<CNetAddr> setKnown;
+    if (setKnown.size() == BITCOIN_TIMEDATA_MAX_SAMPLES)
+        return;
```

Comment:
I had a comment there but have deleted it because it said something like

``` c
// Do not accept more timedata samples (see the comment below).
```

But I have deleted it (it looks so clean with the comment below, that I thought it is better not comment it).
The long comment below should probably just be moved up... What do you think?